fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/FieldKeyExpression.java

```diff
@@ -337,7 +337,7 @@ public boolean equals(Object o) {
             return true;
         }
 
-        if (o == null || getClass() != o.getClass()) {
+        if (o == null || !(o instanceof FieldKeyExpression)) {
```

Comment:
This impersonation by a subclass shouldn't be necessary. Probably another `TODO`.